docs/Release-Process.md

```diff
@@ -0,0 +1,63 @@
+    go mod tidy
+    ```
+in the top-level repo directory, and the `peerpod-ctl` and `volumes/csi-wrapper` directories.
+> **Note:** If there are API changes in the kata-runtime go modules and we need to cloud-api-adaptor to implement, then it may be necessary to temporarily get the peerpod-ctrl and csi-wrapper to self-reference the parent folder to avoid compilation errors. This can be done by running 
+> ```
+> go mod edit -replace github.com/confidential-containers/cloud-api-adaptor=../
+> go mod tidy
+> ```
+> from in the `peerpod-ctrl` and `volumes/csi-wrapper` directories.
+- The attestation-agent that is built into the peer pod vm image, by updating the `AA_VERSION` in the [`Makefile.inc`](../podvm/Makefile.inc)
+
+These updates should be done in a PR that is merged triggering the cloud-api-adaptor [image build workflow](../.github/workflows/image.yaml) to create a new container image in [`quay.io](https://quay.io/repository/confidential-containers/cloud-api-adaptor?tab=tags) to use in testing.
+
+We should also create a cloud-api-adaptor [pre-release](https://github.com/confidential-containers/cloud-api-adaptor/releases/new) to trigger the creation of the podvm build.
+
+These versions should be tested to ensure that there are no breaking changes and the wider confidential-containers release team updated with the status. If there are any issues then this phase might be repeated until it is successful.
+
+### Cutting releases
+
+During this phase the successful release candidates commits are used to cut proper releases for all the components and then the projects that use them updated to point to these releases and re-tested. This shouldn't introduce any instability and all these versions where tested in the release candidate testing phase.
+
+For the cloud-api-adaptor we need to wait until the Kata Containers release tag has been created and the [Kata Containers runtime payload] (https://github.com/kata-containers/kata-containers/actions/workflows/cc-payload.yaml) to have been built. We then can repeat the updates done during the release candidate phase, but this time use the release tags of the projects e.g. `v0.6.0`.
+
+Once this has been completed and merged in we run the latest release of the cloud-api-adaptor including the auto generated release notes.
+
+This will trigger the podvm builds to happen again and we should re-test the release code before updating the confidential-containers release team to let them know it has completed successfully
+
+### Post-release
+
+After the release has been cut the `peerpod-ctrl` and `volumes/csi-wrapper` go modules should be updated to remove any local replace references, and be updated to use the release version of the `cloud-api-adaptor` by running:
+```
+go get github.com/confidential-containers/cloud-api-adaptor
+go mod tidy
+```
+from in the `peerpod-ctrl` and `volumes/csi-wrapper` directories.
+
+## Improvements
```

Comment:
I am wondering if wouldn't be better to cut the release from a release branch, this way we don't need to revert all the changes to go back picking up the latest dependencies after the release. However I don't know how it would create additional problems, for example, having to change some github workflows for CI/release.

Reply:
Yeah, at the moment we effectively just create the release on the main branch and it's frozen forever there with the tag, but there is updates to latest needed. This is pretty similar to the situation with kata, the operator etc where we end up having the released versions baked in for a while post release (and in kata for example we sometimes don't bump image-rs until the next release cycle). Some could argue that this is bad and hides dev incompatibilities, but I guess others could argue that it means we are building on stable versions.

The self-reference with the cloud-api-adaptor and peerpod-ctrl and csi-wrapper is a bit different though, the issue is just that we can't point to a tag that we haven't created yet and the act of updating them, would change the sha. If they were in separate repos it would help the problem, but just defer it to us having more repos to tag and update. It might be interesting to see how this plays out with image-rs, attestation-agent and ocicrypt-rs from now on to see if they have any issues with this.

@yoheiueda - no I recall you saying you had an idea for this?

Reply:
> Yeah, at the moment we effectively just create the release on the main branch and it's frozen forever there with the tag, but there is updates to latest needed. This is pretty similar to the situation with kata, the operator etc where we end up having the released versions baked in for a while post release (and in kata for example we sometimes don't bump image-rs until the next release cycle). Some could argue that this is bad and hides dev incompatibilities, but I guess others could argue that it means we are building on stable versions.
>

Got it. I thought we were reverting to latest just after the release so I thought releasing from a branch would be better. From a CI point of view, I always advocate to build on stable versions :)

> The self-reference with the cloud-api-adaptor and peerpod-ctrl and csi-wrapper is a bit different though, the issue is just that we can't point to a tag that we haven't created yet and the act of updating them, would change the sha. If they were in separate repos it would help the problem, but just defer it to us having more repos to tag and update. It might be interesting to see how this plays out with image-rs, attestation-agent and ocicrypt-rs from now on to see if they have any issues with this.
> 

Definitively I prefer less repos. I don't know Golang enough to propose a better solution (if any exists). Just hoping @yoheiueda got an good idea :)

> @yoheiueda - no I recall you saying you had an idea for this?

Reply:
> I thought we were reverting to latest just after the release so I thought releasing from a branch would be better.

I just to argue the counterpoint - if we start doing the update to overlays and webhook to pin their release versions I proposed in #1109 then that will be quite a few files that we will want to revert to latest soon after release, so maybe the branch is the best approach. My concern is us ended up with a lot of branches that would make things confusing, but I guess it's not disimilar to kata other than the frequency of releases is higher. Definitely an approach to discuss